amespahdbpythonsuite/fitted.py

```diff
@@ -10,4 +19,448 @@ class Fitted(Spectrum):
+                                'operand': keywords.get('small', 50)},
+                      'nitrogen': {'subclass': 'nn',
+                                   'operator': operator.gt,
+                                   'operand': 0}}
+
+        classes = dict()
+
+        for key in subclasses:
+            classes[key] = self.__classes(subclasses[key])
+
+        puids = [uid for uid in self.uids
+                 if self.pahdb['species'][uid]['nn'] == 0
+                 and self.pahdb['species'][uid]['no'] == 0
+                 and self.pahdb['species'][uid]['nmg'] == 0
+                 and self.pahdb['species'][uid]['nsi'] == 0
+                 and self.pahdb['species'][uid]['nfe'] == 0]
+        classes['pure'] = sum({k: v for k, v in self.data.items() if k in puids}.values())
+
+        return classes
+
+    def __classes(self, s):
+        """
+        Retrieves the intensities of a given subclass.
+
+        """
+
+        uids = [uid for uid in self.uids
+                if s['operator'](self.pahdb['species'][uid][s['subclass']], s['operand'])]
+
+        intensities = sum({k: v for k, v in self.data.items() if k in uids}.values())
+        return intensities
+
+    def getbreakdown(self, **keywords):
+        """
+        Retrieves the breakdown of the fitted PAHs.
+
+        """
+        if not self.pahdb:
+            message('VALID DATABASE NEEDED TO GET CLASSES')
+
+            return None
+
+        # Set atom keywords in species dictionary.
+        self._atoms()
+
+        # Getting fit weights
+        fweight = np.array(list(self.weights.values()))
+
+        breakdown = {'solo': np.sum([self.pahdb['species'][uid]['n_solo'] for uid in self.uids]),
+                     'duo': np.sum([self.pahdb['species'][uid]['n_duo'] for uid in self.uids]),
+                     'trio': np.sum([self.pahdb['species'][uid]['n_trio'] for uid in self.uids]),
+                     'quartet': np.sum([self.pahdb['species'][uid]['n_quartet'] for uid in self.uids]),
+                     'quintet': np.sum([self.pahdb['species'][uid]['n_quintet'] for uid in self.uids])
+                     }
+
+        total = 1.0
+
+        if keywords.get('flux', False):
+            if not self.grid:
+                message('GRID IS NOT SET')
+
+                return None
+
+            classes = self.getclasses(**keywords)
+
+            if not keywords.get('absolute', False):
+                total, err = integrate.quad(self.grid, self.getfit())
+                print('first integrate occurance')
+
+            for key in classes:
+                breakdown[key] = integrate.quad(self.grid, classes[key]) / total
+                print('second integrate occurance')
+
+            return breakdown
+
+        if not self.weights:
+            message('WEIGHTS ARE NOT SET')
+
+            return None
+
+        if not keywords.get('absolute', False):
+            total = np.sum(fweight)
+
+        subclasses = {'anion': {'subclass': 'charge',
```

Comment:
This is a repeat of what is at 286; can it put in a method or made a class constant?